src/index.web.tsx

```diff
@@ -0,0 +1,122 @@
+type Profiler = any;
+declare global {
+  interface Window {
+    Profiler: Profiler;
+  }
+}
+
+const NOT_STARTED_PROFILER = {
+  stop: (): Promise<ProfilingTrace> => {
+    console.warn('The profiler has not been started');
+
+    return Promise.resolve({
+      frames: [],
+      resources: [],
+      samples: [],
+      stacks: [],
+    });
+  },
+};
+let profiler = NOT_STARTED_PROFILER;
+
+const SAMPLING_INTERVAL = 10;
+const MAX_BUFFER_SIZE = 10000;
+const MICRO_SECONDS_IN_MILLISECONDS = 1000;
+
+export function startProfiling(): boolean {
+  if (typeof window.Profiler !== 'function') {
+    console.warn(
+      'Profiler is not available in this browser. Ignoring this profiler run...'
+    );
+
+    return false;
+  }
+
+  profiler = new Profiler({
+    sampleInterval: SAMPLING_INTERVAL,
+    maxBufferSize: MAX_BUFFER_SIZE,
+  });
+
+  return true;
+}
+
+export async function stopProfiling(saveToDownloads = false): Promise<string> {
+  const trace = await profiler.stop();
+
+  profiler = NOT_STARTED_PROFILER;
+
+  if (saveToDownloads) {
+    downloadJsonFile(
+      convertToHermesProfilerFormat(trace),
+      `trace-${new Date().toISOString()}`
+    );
```

Comment:
NIT: can we separate the statements?

const traceJson = convert...
const traceName = ...
downloadJsonFile